### PR TITLE
fix: move emptiness tests to deprovisioning and deduplicate deletion logic

### PR DESCRIPTION
**Pull request by njtran**
*Issue #, if available:*

*Description of changes:*
- removes duplicated deprovisioning logic used for emptiness. `node/emptiness.go` and `deprovisioning/controller.go` were both executing deletion logic. 
- moves respective tests over

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/controllers/deprovisioning/suite_test.go [hunk 1]
```diff
@@ -1276,7 +1276,7 @@ var _ = Describe("Topology Consideration", func() {
 })
 
 var _ = Describe("Empty Nodes", func() {
-	It("can delete empty nodes", func() {
+	It("can delete empty nodes with Consolidaiotn", func() {
```
Comment (jonathan-innis) [inline pkg/controllers/deprovisioning/suite_test.go]:
```suggestion
	It("can delete empty nodes with Consolidation", func() {
```

File: pkg/controllers/node/emptiness.go [hunk 2]
```diff
@@ -81,19 +80,7 @@ func (r *Emptiness) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisi
 		logging.FromContext(ctx).Infof("Added TTL to empty node")
 		return reconcile.Result{RequeueAfter: ttl}, nil
```
Comment (jonathan-innis) [inline pkg/controllers/node/emptiness.go:81]: No need to requeue after the ttl now, right?

File: pkg/controllers/node/emptiness.go [hunk 3]
```diff
@@ -65,7 +64,7 @@ func (r *Emptiness) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisi
 		return reconcile.Result{}, nil
 	}
 
-	emptinessTimestamp, hasEmptinessTimestamp := n.Annotations[v1alpha5.EmptinessTimestampAnnotationKey]
+	_, hasEmptinessTimestamp := n.Annotations[v1alpha5.EmptinessTimestampAnnotationKey]
 	if !empty {
```
Comment (jonathan-innis) [inline pkg/controllers/node/emptiness.go:68]: I think this whole thing can be compressed into

```go
if !empty && hasEmptinessTimestamp {
  delete(n.Annotations, v1alpha5.EmptinessTimestampAnnotationKey)
  logging.FromContext(ctx).Infof("Removed emptiness TTL from node")
} else if empty && !hasEmptinessTimeStamp {
  n.Annotations[v1alpha5.EmptinessTimestampAnnotationKey] = r.clock.Now().Format(time.RFC3339)
  logging.FromContext(ctx).Infof("Added TTL to empty node")
}
return reconcile.Result{}, nil
```

Comment (njtran) [reply pkg/controllers/node/emptiness.go:68]: Good idea! Let me condense it.

File: pkg/controllers/node/emptiness.go [hunk 4]
```diff
@@ -65,35 +62,17 @@ func (r *Emptiness) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisi
...
-	// 3. Set TTL if not set
-	n.Annotations = lo.Assign(n.Annotations)
-	ttl := time.Duration(ptr.Int64Value(provisioner.Spec.TTLSecondsAfterEmpty)) * time.Second
-	if !hasEmptinessTimestamp {
+	_, hasEmptinessTimestamp := n.Annotations[v1alpha5.EmptinessTimestampAnnotationKey]
+	if !empty && hasEmptinessTimestamp {
+		delete(n.Annotations, v1alpha5.EmptinessTimestampAnnotationKey)
+		logging.FromContext(ctx).Infof("Removed emptiness TTL from node")
+	} else if empty && !hasEmptinessTimestamp {
+		n.Annotations = lo.Assign(n.Annotations)
```
Comment (jonathan-innis) [inline pkg/controllers/node/emptiness.go:70]: This line doesn't do anything now

Comment (njtran) [reply pkg/controllers/node/emptiness.go:70]: Pretty sure it panics if n.Annotations is nil. Check this. https://gist.github.com/chakrit/8157474#file-main-go-L15

Comment (jonathan-innis) [reply pkg/controllers/node/emptiness.go:70]: Ah, good call. Why not?

```go
n.Annotations = lo.Assign(n.Annotations, map[string]string{
  v1alpha5.EmptinessTimestampAnnotationKey: r.clock.Now().Format(time.RFC3339)
})
```

**Review by jonathan-innis — APPROVED** [review]

File: pkg/controllers/node/emptiness.go (hunk 4 above)
Comment (njtran) [reply pkg/controllers/node/emptiness.go:70]: Sure can do

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**